<launch>
  <include file="$(find pr2_moveit_config)/launch/planning_context.launch" />

  <node pkg="humanoid_catching" type="catch_human_action_server" name="catch_human_action">
    <param name="urdf" value="$(find humanoid_catching)/models/pr2.urdf" />
    <param name="max_distance" value="0.821000"/>
    <param name="left_arm_data" value="$(find kinematics_cache)/octree/left_arm.ot" />
    <param name="right_arm_data" value="$(find kinematics_cache)/octree/right_arm.ot" />
  </node>

  <node pkg="humanoid_catching" type="fall_predictor" name="fall_predictor"/>

  <node pkg="humanoid_catching" type="imu_filter" name="imu_filter">
    <remap to="/human/imu" from="/in" />
    <remap to="/human/imu_filtered" from="/out" />
  </node>

  <node pkg="humanoid_catching" type="catching_controller" name="catching_controller">
    <remap to="/human/imu_filtered" from="/in" />
  </node>

  <node pkg="humanoid_catching" type="balancer" name="balancer"/>
  <node pkg="humanoid_catching" type="fall_stats_recorder" name="fall_stats_recorder"/>
</launch>
